pkg/fqdn/helpers_test.go

```diff
@@ -37,3 +41,68 @@ func (ds *DNSCacheTestSuite) TestKeepUniqueNames(c *C) {
 		c.Assert(val, checker.DeepEquals, item.expected)
 	}
 }
+
+func (ds *DNSCacheTestSuite) TestInjectCIDRSetRulesWithOtherCIDRSet(c *C) {
+	// Validate that if empty cache the ToCidrRule is always empty
+	rule := makeRule("cilium.io")
+	cache := NewDNSCache()
+	cache.Update(now, "cilium.io.", []net.IP{net.ParseIP("1.1.1.1")}, 1)
+	rule.Egress[0].ToFQDNs = api.FQDNSelectorSlice{
+		{MatchName: "cilium.io"},
+	}
+	rule.Egress = append(rule.Egress, api.EgressRule{ToCIDRSet: api.IPsToCIDRRules([]net.IP{net.ParseIP("4.4.4.4")})})
+
+	injectToCIDRSetRules(rule, cache, nil)
+	c.Assert(rule.Egress[0].ToCIDRSet, HasLen, 1)
+	c.Assert(rule.Egress[1].ToCIDRSet, HasLen, 1)
+}
+
+func (ds *DNSCacheTestSuite) TestInjectCIDRSetRulesInvalidCache(c *C) {
+	// Validate that if empty cache the ToCidrRule is always empty
+	rule := makeRule("cilium.io")
+	rule.Egress[0].ToFQDNs = api.FQDNSelectorSlice{
+		{MatchName: "cilium.io"},
+	}
+	EmptyCache := NewDNSCache()
+	injectToCIDRSetRules(rule, EmptyCache, nil)
+	c.Assert(rule.Egress[0].ToCIDRSet, HasLen, 0)
+
+	// Validate that if empty cache the ToCidrRule is cleared correctly
+	rule.Egress[0].ToCIDRSet = api.IPsToCIDRRules([]net.IP{net.ParseIP("1.1.1.1")})
+	c.Assert(rule.Egress[0].ToCIDRSet, HasLen, 1)
+	injectToCIDRSetRules(rule, EmptyCache, nil)
+	c.Assert(rule.Egress[0].ToCIDRSet, HasLen, 0)
+}
+
+func (ds *DNSCacheTestSuite) TestInjectCIDRSetRulesByMatchName(c *C) {
+	rule := makeRule("cilium.io", "cilium.io")
+	cache := NewDNSCache()
+	cache.Update(now, "cilium.io.", []net.IP{net.ParseIP("1.1.1.1")}, 1)
+
+	injectToCIDRSetRules(rule, cache, nil)
+	c.Assert(rule.Egress[0].ToCIDRSet, HasLen, 1)
+}
+
+func (ds *DNSCacheTestSuite) TestInjectCIDRSetRulesByMatchPattern(c *C) {
+
+	rule := makeRule("cilium.io")
+	rule.Egress[0].ToFQDNs = api.FQDNSelectorSlice{
+		{MatchPattern: "cilium.io"},
+	}
+	cache := NewDNSCache()
+	cache.Update(now, "cilium.io.", []net.IP{net.ParseIP("1.1.1.1")}, 1)
+	regexp := regexpmap.NewRegexpMap()
+	regexp.Add("cilium.io", "cilium.io")
```

Comment:
it might be good to use matchpattern.PreparePattern (or whatever it's called) to be more consistent with the normal usage. It doesn't matter too much, though.